src/presign/participant.rs

```diff
@@ -947,55 +952,62 @@ pub(crate) struct PresignKeyShareAndInfo {
 }
 
 impl PresignKeyShareAndInfo {
-    /// Corresponds to pre-signing round 1 for party i
-    ///
-    /// Produces local shares k and gamma, along with their encrypted
-    /// components K = enc(k) and G = enc(gamma).
+    fn new(id: ParticipantIdentifier, input: &Input) -> Result<Self> {
+        Ok(Self {
+            aux_info_private: input.auxinfo_private.clone(),
+            aux_info_public: input.find_auxinfo_public(id)?.clone(),
+            keyshare_private: input.keyshare_private.clone(),
+            keyshare_public: input.find_keyshare_public(id)?.clone(),
+        })
+    }
+
+    /// Round one of the presign protocol.
     ///
-    /// The public_keys parameter corresponds to a KeygenPublic for
-    /// each of the other parties.
+    /// This round produces two local shares `k` and `ɣ` along with their
+    /// ciphertexts encrypted using this participant's Paillier encryption key,
+    /// alongside a zero knowledge proof that the ciphertext associated with `k`
+    /// was constructed correctly. We utilize the `other_auxinfos` argument to
+    /// construct this proof, as it has to be tailored to the particular
+    /// receiving participant.
     #[cfg_attr(feature = "flame_it", flame("presign"))]
-    pub(crate) fn round_one<R: RngCore + CryptoRng>(
+    fn round_one<R: RngCore + CryptoRng>(
         &self,
         rng: &mut R,
         context: &impl ProofContext,
-        public_keys: &[&AuxInfoPublic],
+        other_auxinfos: &[&AuxInfoPublic],
     ) -> Result<(
         RoundOnePrivate,
         HashMap<ParticipantIdentifier, RoundOnePublic>,
         RoundOnePublicBroadcast,
     )> {
         let order = k256_order();
 
-        // Sample k <- F_q
         let k = random_positive_bn(rng, &order);
-        // Sample gamma <- F_q
         let gamma = random_positive_bn(rng, &order);
 
-        // Sample rho <- Z_N^* and set K = enc(k; rho)
         let (K, rho) = self
             .aux_info_public
             .pk()
             .encrypt(rng, &k)
             .map_err(|_| InternalError::InternalInvariantFailed)?;
-        // Sample nu <- Z_N^* and set G = enc(gamma; nu)
         let (G, nu) = self
             .aux_info_public
             .pk()
             .encrypt(rng, &gamma)
             .map_err(|_| InternalError::InternalInvariantFailed)?;
-
         let mut r1_publics = HashMap::new();
-        for aux_info_public in public_keys {
-            // Compute psi_{j,i} for every participant j != i
+        let secret = crate::zkp::pienc::PiEncSecret::new(k.clone(), rho.clone());
```

Comment:
suggestion: can we import this type and use plain `PiEncSecret` here (and for `PiEncInput` below)?